### PR TITLE
fix bigint serializing

### PR DESCRIPTION
**Pull request by zl730**
# PR Details

## Description

bug fix on bigint serializing

## How Has This Been Tested

Unit tests passed

### CODE REVIEW

File: src/main/scala/vsys/blockchain/state/opcdiffs/BasicOpcDiff.scala [hunk 1]
```diff
@@ -128,6 +128,14 @@ object BasicOpcDiff extends OpcDiffer {
     val Not         = BasicTypeVal(14, 3, Seq(1),    (b, d) => not(d(b(1))))
   }
 
+  private def bigintBytes (bytes: Array[Byte]): Array[Byte] = {
+    val x = bytes.dropRight(1).dropWhile(i => i == 0) ++ bytes.takeRight(1)
```
Comment (twwu123) [inline src/main/scala/vsys/blockchain/state/opcdiffs/BasicOpcDiff.scala:132]: This function seems to have the same effect as BigInt(bytes).toByteArray

Comment (ncying) [reply src/main/scala/vsys/blockchain/state/opcdiffs/BasicOpcDiff.scala:132]: it seems that BigInt apply java BigInteger to transfer the format
```scala
  /** Translates a byte array containing the two's-complement binary
   *  representation of a BigInt into a BigInt.
   */
  def apply(x: Array[Byte]): BigInt =
    new BigInt(new BigInteger(x))
```

Based on the BigInteger source code(http://hg.openjdk.java.net/jdk/jdk11/file/tip/src/java.base/share/classes/java/math/BigInteger.java), it seems that we can use
```scala
if(BigInt(bytes).isValidInt) Ints.toByteArray(BigInt(bytes).toInt)
if(BigInt(bytes).isValidLong) Longs.toByteArray(BigInt(bytes).toInt)

```
here?
@zl730 will it causes extra calculation here?

Comment (zl730) [reply src/main/scala/vsys/blockchain/state/opcdiffs/BasicOpcDiff.scala:132]: Pure byte array operations here avoiding BigInt object creation for lower time and space complexity

**Review by twwu123 — APPROVED** [review]

File: src/main/scala/vsys/blockchain/state/opcdiffs/BasicOpcDiff.scala [hunk 2]
```diff
@@ -143,18 +143,11 @@ object BasicOpcDiff extends OpcDiffer {
   }
 
   private def bigintBytes (bytes: Array[Byte]): Array[Byte] = {
```
[the remark this answers is not in the document]
Comment (zl730) [reply src/main/scala/vsys/blockchain/state/opcdiffs/BasicOpcDiff.scala]: this is fixed by recent commit

**Pull request merged**